Consul/Connect.cs

```diff
@@ -297,6 +302,82 @@ public Task<QueryResult<List<ServiceIntention>>> ListIntentions<ServiceIntention
             var req = _client.Get<List<ServiceIntention>>("/v1/connect/intentions", q);
             return req.Execute(ct);
         }
+
+        /// <summary>
+        /// Creates a new intention
+        /// </summary>
+        /// <param name="intention"></param>
+        /// <param name="q"></param>
+        /// <param name="ct"></param>
+        /// <returns>Returns the ID of the created intention.</returns>
+        public async Task<WriteResult<ServiceIntentionCreateResponse>> CreateIntentionWithID(ServiceIntention intention, WriteOptions q, CancellationToken ct = default)
+        {
+            return await _client.Post<ServiceIntention, ServiceIntentionCreateResponse>("v1/connect/intentions", intention, q).Execute(ct).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Creates a new intention
+        /// </summary>
+        /// <param name="intention"></param>
+        /// <param name="ct"></param>
+        /// <returns>Returns the ID of the created intention</returns>
+        public Task<WriteResult<ServiceIntentionCreateResponse>> CreateIntentionWithID(ServiceIntention intention, CancellationToken ct = default)
+        {
+            return CreateIntentionWithID(intention, WriteOptions.Default, ct);
+        }
+
+        /// <summary>
+        /// Reads a specific intention by ID.
+        /// </summary>
+        /// <typeparam name="ServiceIntention"></typeparam>
+        /// <param name="uuid"></param>
+        /// <param name="ct"></param>
+        /// <returns>A service intention</returns>
+        public Task<QueryResult<ServiceIntention>> ReadIntentionByID<ServiceIntention>(string uuid, CancellationToken ct = default)
+        {
+            return ReadIntentionByID<ServiceIntention>(uuid, QueryOptions.Default, ct);
+        }
+
+        /// <summary>
+        /// Reads a specific intention by ID.
+        /// </summary>
+        /// <typeparam name="ServiceIntention"></typeparam>
+        /// <param name="uuid"></param>
+        /// <param name="q"></param>
+        /// <param name="ct"></param>
+        /// <returns>A service intention</returns>
+        public Task<QueryResult<ServiceIntention>> ReadIntentionByID<ServiceIntention>(string uuid, QueryOptions q, CancellationToken ct = default)
+        {
+            var req = _client.Get<ServiceIntention>($"v1/connect/intentions/{uuid}", q);
+            return req.Execute(ct);
+        }
+
+        /// <summary>
+        /// creates a new intention 
```

Comment:
You can include in the comment the information that this endpoint doesn't let you assign all the fields (see the docs -> https://developer.hashicorp.com/consul/api-docs/v1.11.x/connect/intentions#upsert-intention-by-name).